### PR TITLE
chore: bump CI nighlty + remove `=` dependency

### PR DESCRIPTION
**Pull request by carllerche**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ log      = "0.4"
 slab     = "0.4.0"
 net2     = "0.2.29"
 iovec    = "0.1.1"
-cfg-if   = "=0.1.9"
```
Comment (hawkw) [inline Cargo.toml]: was there a reason for the “=“ dep, or is this just lost to the ages?

Comment (carllerche) [reply Cargo.toml]: no idea.

Comment (carllerche) [reply Cargo.toml]: Oh, to match the min rust version  mio targets.

Comment (carllerche) [reply Cargo.toml]: I reverted to keep it simple.

**Pull request merged**